rfcs/0145-doc-strings.md

```diff
@@ -119,19 +117,157 @@ In the following we give a comprehensive overview to our decision that we've mad
 
-# Decisions
+> Note: Research of the RFC Sheperds Team showed That this allows for intuitive placements like are already done in nixpkgs. 
+
+### Lambda Examples
+
+```nix
+/**Doc for anonymous lambda function*/
+↓
+x: x;
+
+# doc -> Lambda Value '(x: x)'
+```
+
+```nix
+/**Doc for assigned lambda function*/
+↓
+assigned = x: x;
+```
+
+```nix
+/**Doc B*/
+assigned = /**Doc A*/x: x;
+
+# Lambda Documentation is 'Doc A' because it is directly next to the documentable lambda.
+```
+
+### Attribute Examples
+
+```nix
+{
+    /** Documentation for 'foo' */
+    ↓ 
+    foo = 1;
+}
+# doc -> Attribute Name foo
+```
+
+```nix
+{
+    /** Documentation for 'bar' */
+    ↓
+    ${let name = "bar"; in name} = 2;
+}
+# Interpolated attribute names
+```
+
+```nix
+{
+    /** Documentation for 'bar' */
+    ↓
+    ${let name = "bar"; in name} = 2;
+}
+# Interpolated attribute names
+```
+
+```nix
+setWithMappedNamed =
+    lib.mapAttrs' (name: value: {
+      /**Undocumentable attribute name*/
+              ↓
+      name = "${n + "foo"}";
+      inherit value;
+    })
+    {
+      a = 1;
+      b = 2;
+    };
+# Dynamically constructed attribute names cannot be documented. (constructed in a function call like above)
+``` 
+
+### Examples when the lambda is not directly assigned
+
+```nix
+{
+    /**This documents (1), NOT (2) */
+    ↓1    ↓2
+    foo = map (x: x);
+          ^~~~~~~~~~~ Evaluates to a specialisation of map, which is a partially applied lambda and defined elsewhere.
+          
+# IMPORTANT: Documentation of the specialized map function can still be done as shown. But is related to attrName 'foo' (1)
+}
+```
+
+```nix
+{
+    /**This documents (1), NOT (2) */
+    ↓1                           ↓2
+    foo = mapAttrs (name: value: (x: x) ) {a=1; b=2;};
+    # => 
+    # foo = {
+    #   a = x: x;
+    #   b = x: x; 
+    # }
+# Doesn't document the inner anonymous lambda function.
+}
+```
+
+### Partially applied lambda
+
+All partial functions can share the same placement with the outermost lambda.
+
+```nix
+/**Generic doc for all partial lambda functions too*/
+↓1 ↓2          ↓3
+x: ({y, ...}:  z: x + y) * z;
+
+# 1 -> Lambda Value `x: ({y, ...}: z: x + y) * z` countApplied: 0
+# 2 -> Lambda Value `({y, ...}: z: x + y) * z`    countApplied: 1
+# 3 -> Lambda Value `z: x + y) * z`               countApplied: 2
+```
 
+```nix
+{
+    /** incrMap. Adds one to every item. */
+    ↓    
+    incrMap = map (x: x + 1);
+              ^~~~~~~~~~~ Evaluates to a specialisation of map, which is a partially applied lambda and defined elsewhere.
+              
+# The lambda documementation is taken from the place where the `map` lambda is defined
+# Retrieve the documentation from the attribute 'incrMap'.
+}
+```
+
+Still all other 'Fundamentals' apply.
+
+```nix
+                                 Always doc for anonymous lambda function 'y: x + y'
+                                 ↓    
+x: /** Partial Doc for y: x+y */ y: x + y;
```

Comment:
Is there any example showing how to access the anonymous lambda and get it's doc? I think it is somehow unnecessary to document anonymous lambdas actually.

Reply:
Yes. There are some examples in nixpkgs and i think it is important to support documentation in the same file, where the implementation is. 
e.g.
https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix